controllers/host/host_controller_test.go

```diff
@@ -352,4 +356,725 @@ var _ = Describe("Host controller", func() {
+					{
+						Name: "backup",
+						Size: 1,
+					},
+				}},
+			}
+			other := &starlingxv1.HostProfileSpec{
+				Storage: &starlingxv1.ProfileStorageInfo{Monitor: nil, OSDs: nil, VolumeGroups: nil, FileSystems: &starlingxv1.FileSystemList{
+					{
+						Name: "backup",
+						Size: 1,
+					},
+				}},
+			}
+
+			var k8sManager, _ = ctrl.NewManager(cfg, ctrl.Options{
+				Scheme:             scheme.Scheme,
+				MetricsBindAddress: "0",
+			})
+			r := &HostReconciler{
+				Client: k8sManager.GetClient(),
+				Scheme: k8sManager.GetScheme(),
+			}
+
+			got := r.CompareFileSystemTypes(in, other)
+			Expect(got).To(BeTrue())
+		})
+		It("Should return true when in and other storage is not equal and have unallowed filesystem types", func() {
+			in := &starlingxv1.HostProfileSpec{
+				Storage: &starlingxv1.ProfileStorageInfo{Monitor: nil, OSDs: nil, VolumeGroups: nil, FileSystems: &starlingxv1.FileSystemList{
+					{
+						Name: "backup",
+						Size: 5,
+					},
+					{
+						Name: "docker",
+						Size: 2,
+					},
+				}},
+			}
+			other := &starlingxv1.HostProfileSpec{
+				Storage: &starlingxv1.ProfileStorageInfo{Monitor: nil, OSDs: nil, VolumeGroups: nil, FileSystems: &starlingxv1.FileSystemList{
+					{
+						Name: "backup",
+						Size: 1,
+					},
+					{
+						Name: "docker",
+						Size: 3,
+					},
+				}},
+			}
+
+			var k8sManager, _ = ctrl.NewManager(cfg, ctrl.Options{
+				Scheme:             scheme.Scheme,
+				MetricsBindAddress: "0",
+			})
+			r := &HostReconciler{
+				Client: k8sManager.GetClient(),
+				Scheme: k8sManager.GetScheme(),
+			}
+
+			got := r.CompareFileSystemTypes(in, other)
+			Expect(got).To(BeTrue())
+		})
+		It("Should return false when in and other storage is not equal and have allowed filesystem types", func() {
+			in := &starlingxv1.HostProfileSpec{
+				Storage: &starlingxv1.ProfileStorageInfo{Monitor: nil, OSDs: nil, VolumeGroups: nil, FileSystems: &starlingxv1.FileSystemList{
+					{
+						Name: "instances",
+						Size: 5,
+					},
+				}},
+			}
+			other := &starlingxv1.HostProfileSpec{
+				Storage: &starlingxv1.ProfileStorageInfo{Monitor: nil, OSDs: nil, VolumeGroups: nil, FileSystems: &starlingxv1.FileSystemList{
+					{
+						Name: "instances",
+						Size: 2,
+					},
+					{
+						Name: "ceph",
+						Size: 1,
+					},
+				}},
+			}
+
+			var k8sManager, _ = ctrl.NewManager(cfg, ctrl.Options{
+				Scheme:             scheme.Scheme,
+				MetricsBindAddress: "0",
+			})
+			r := &HostReconciler{
+				Client: k8sManager.GetClient(),
+				Scheme: k8sManager.GetScheme(),
+			}
+
+			got := r.CompareFileSystemTypes(in, other)
+			Expect(got).To(BeFalse())
+		})
+	})
+
+	Context("CompareOSDs func", func() {
+		It("SHould return false when other hostProfileSpec is nil", func() {
+			in := &starlingxv1.HostProfileSpec{}
+			other := &starlingxv1.HostProfileSpec{}
```

Comment:
Try creating instance with 'var' keyword

Reply:
Updated to the var